pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,429 @@
+            # link axis to view
+            axis = self.axes[axis_name]
+            # connect view changes to axis
+            # set axis main view link if not assigned
+            if axis.linkedView() is None:
+                axis._linkedView = weakref.ref(chart_vb)
+            # FROM AxisItem.linkToView
+            # connect view changes to axis changes
+            if axis.orientation in ["right", "left"]:
+                signals["chart_vb.sigYRangeChanged"] = chart_vb.sigYRangeChanged.connect(
+                    axis.linkedViewChanged)
+            elif axis.orientation in ["top", "bottom"]:
+                signals["chart_vb.sigXRangeChanged"] = chart_vb.sigXRangeChanged.connect(
+                    axis.linkedViewChanged)
+            signals["chart_vb.sigResized"] = chart_vb.sigResized.connect(
+                axis.linkedViewChanged)
+            axis_view = axis.linkedView()
+            if chart_vb is not axis_view:
+                # FROM ViewBox.linkView
+                # connext axis's view changes to view since axis acts just like a proxy to it
+                if axis.orientation in ["top", "bottom"]:
+                    # connect axis main view changes to view
+                    chart_vb.state["linkedViews"][chart_vb.XAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    signals["axis_view.sigXRangeChanged"] = axis_view.sigXRangeChanged.connect(
+                        chart_vb.linkedXChanged)
+                    signals["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        chart_vb.linkedXChanged)
+                    # disable autorange on manual movements
+                    signals["axis_view.sigXRangeChangedManually"] = axis_view.sigXRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
+                elif axis.orientation in ["right", "left"]:
+                    # connect axis main view changes to view
+                    chart_vb.state["linkedViews"][chart_vb.YAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    signals["axis_view.sigYRangeChanged"] = axis_view.sigYRangeChanged.connect(
+                        chart_vb.linkedYChanged)
+                    signals["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        chart_vb.linkedYChanged)
+                    # disable autorange on manual movements
+                    signals["axis_view.sigYRangeChangedManually"] = axis_view.sigYRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
```

Comment:
signal connected to a lambda function that references `self`

Reply:
added workaround